### PR TITLE
Mods to the recoder. Input to ease changing the length of the audio file. Date stamped to the wav filename

### PR DESCRIPTION
**Pull request by EA5JQP**
I did some modifications on the frontend for the Recorder:
1- Added an input to ease changing the length of the audio file. With the slider we can go up to 10 min, but up to the maximum can be introduce in the input field.
2- In order to avoid rewriting the same audio file, the date and time is added to the wav filename.

### CODE REVIEW

File: frontend/src/stores/recorder.ts [hunk 1]
```diff
@@ -3,14 +3,18 @@ import { ApiPaths, ApiResponse } from "../types/Api";
 import { Notify } from "quasar";
 import axios from "axios";
 import { Recorder } from "../types/Recorder";
+import { date } from 'quasar'
 
 const axiosInstance = axios.create();
 axiosInstance.defaults.timeout = 600;
 
+const timeStamp = Date.now()
+const formattedTimeStamp = date.formatDate(timeStamp, 'YYYYMMDD_HHmmss')
```
Comment (kamilsss655) [inline frontend/src/stores/recorder.ts:12]: Can you extract `formattedTimeStamp` logic to some helper file so implementation is not repeated across different files?

File: frontend/package-lock.json [hunk 2]
```diff
@@ -836,12 +836,13 @@
       }
     },
     "node_modules/braces": {
-      "version": "3.0.2",
-      "resolved": "https://registry.npmjs.org/braces/-/braces-3.0.2.tgz",
-      "integrity": "sha512-b8um+L1RzM3WDSzvhm6gIz1yfTbBt6YTlcEKAvsmqCZZFw46z626lVj9j1yEPW33H5H+lBQpZMP1k8l+78Ha0A==",
```
Comment (kamilsss655) [inline frontend/package-lock.json:841]: I don't think this PR requires changes to `frontend/package-lock.json`

**Review by kamilsss655 — COMMENTED** [review]
LGTM, left some comments.

**Comment by EA5JQP** [comment]
I made the suggested changes. Thanks a lot!

File: frontend/src/helpers/Time.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { date } from 'quasar';
+
+export const useFormattedTimeStamp = (): string => {
```
Comment (kamilsss655) [inline frontend/src/helpers/Time.ts:3]: Good! Couple of remarks:

1. Let's refactor this function so it accepts `Date` parameter, and at call sites pass `Date.now()` as a param to it.  That way we can use it in the future for formatting other dates.
2. Maybe rename the function to better describe it's function i.e `formatDate` or `formatTimestamp` ?

**Review by kamilsss655 — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/Time.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { date } from 'quasar';
+
+export const formatTimestamp  = (dateInput: Date): string => {
```
Comment (kamilsss655) [inline frontend/src/helpers/Time.ts:3]: :+1:

**Review by kamilsss655 — CHANGES_REQUESTED** [review]
The code looks good.

I did test it now and found a bug where user can enter a higher value, but the slider will display the max of 600. In terms of user experience this is confusing because now there are 2 values displayed and user will not know which one is going to be respected.
image

To remedy this I'd suggest a different approach where there is only a slider, but a user can select units of:
* seconds
* minutes
* hours

This would solve granularity problem without causing user confusion.

What do you think?

**Comment by EA5JQP** [comment]
Yes, I agree it is confussing. My initial idea was to limit the slider up to 10 min to make it simpler, but still keeping the possibility of editing via the q-input. A quick "fix" would be to remove the value in the slider itself (600 in your screenshot) since it is already shown in the q-input.

Regarding one slider and three units, will the user be able to switch units? So if I want 1h 30min 40sec I would have move the slider and switch the unit a couple of times? In this case, I would use directly three different sliders.

**Comment by kamilsss655** [comment]
I like your three sliders idea!

**Comment by EA5JQP** [comment]
Now we have three sliders. Somehow @input didn't update the value automatically when moving the slider, but @change works as well. Probably there is much room for improvement on the presentation side.

**Comment by kamilsss655** [comment]
I like the look with 3 sliders!

Couple of suggestions:
* maybe consider adding `label-always` to sliders so the user see the current value (total duration in seconds in not that comprehensive to humans, especially if the user selected couple of hours)
* also when I select some values with a slider and leave the page and then come back they are initialized as 0
  * perhaps we should move seconds, minutes, hours to `store` so they are kept in between user navigating to different pages?

**Comment by EA5JQP** [comment]
Thank you for the suggestions!

Here are the changes:
- I like how `label-always` show the current value, so I removed the `div` with the total number of seconds. 
- The value of seconds, minutes and hours are now stored. For this I created a new file: stores/duration.ts

**Comment by kamilsss655** [comment]
I've done some refactoring and cleanup and merged it in. Should be in next release!

**Pull request merged**